File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const anagramMap = {};
    for (let i = 0; i < strs.length; i++) {
        const sortedWord = strs[i].split('').sort().join('');
        if (anagramMap[sortedWord]) {
            anagramMap[sortedWord].push(strs[i])
        } else {
            anagramMap[sortedWord] = [strs[i]]
        }
    }
    const finalArray = [];
    for(const anagram in anagramMap) {
        finalArray.push(anagramMap[anagram]);
    }
    return finalArray;
};

File: 49-group-anagrams/group-anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    strs_hash = {}
    strs.each do |str|
        sorted_str = str&.chars&.sort()
        strs_hash[sorted_str] = [] if strs_hash[sorted_str] == nil
        strs_hash[sorted_str].push(str)
    end
    strs_array = strs_hash.values
    return strs_array
end